lib/request-handler.js

```diff
@@ -78,8 +77,8 @@ RequestHandler.prototype.iterateThroughHosts = function (iterator, callback) {
       return callback(new errors.NoHostAvailableError(this.triedHosts));
     }
     host = item.value;
-    // set the distance relative to the client first, using the load balancing policy
-    var distance = host.getDistance(this.loadBalancingPolicy);
+    // set the distance relative to the client first
+    var distance = this.profileManager.getDistance(host);
```

Comment:
That makes sense.  I was conflating distance with host selection (newQueryPlan).  I think `this.loadBalancingPolicy` being used for the query plan and `profileManager.getDistance` being used for distance makes sense.  Especially given your point about how that is used for determining pool size.